src/Microsoft.SqlTools.ServiceLayer/ShowPlan/Contracts/ExecutionPlanGraph.cs

```diff
@@ -113,12 +117,39 @@ public class ExecutionPlanEdges
         /// <summary>
         /// Size of the rows returned by the subtree of the edge.
         /// </summary>
-        /// <value></value>
         public double RowSize { get; set; }
         /// <summary>
         /// Edge properties to be shown in the tooltip.
         /// </summary>
-        /// <value></value>
         public List<ExecutionPlanGraphPropertyBase> Properties { get; set; }
     }
+
+
+    public class ExecutionPlanRecommendation
+    {
+        /// <summary>
+        /// Text displayed in the show plan graph control
+        /// </summary>
+        public string DisplayString { get; set; }
+        /// <summary>
+        /// Raw query that is recommended to the user
+        /// </summary>
+        public string Query { get; set; }
+        /// <summary>
+        /// Query that will be opened in a new file once the user click on the recommendation
+        /// </summary>
+        public string QueryWithDescription { get; set; }
+    }
+
+    public class RawExecutionPlanGraph 
+    {
+        /// <summary>
+        /// Raw execution plan graph from which the query plan graph is generated
+        /// </summary>
+        public string RawGraphString { get; set; }
+        /// <summary>
+        /// file type for raw execution plan. This will be the file type of the editor when the user opens the raw graph
+        /// </summary>
+        public string RawGraphFileType { get; set; }
```

Comment:
This can just be `GraphFileType` I would think

Reply:
I would remove `Raw` from all the names as it isn't really necessary, IMO.